pkg/nodeagent/controller/operatingsystemconfig/reconciler.go

```diff
@@ -103,6 +103,9 @@ type Reconciler struct {
 	HostName      string
 	NodeName      string
 	MachineName   string
+	// SkipWriteLastAppliedConfiguration is used by gardenadm that it does not confuse gardener-node-agent's change
+	// detection when applying other OSCs than the original one.
+	SkipWriteLastAppliedConfiguration bool
```

Comment:
I am somewhat confused by the comment.

```suggestion
	// SkipWriteLastAppliedConfiguration is used by gardenadm to not confuse gardener-node-agent's change
	// detection when applying other OSCs than the original one.
	SkipWriteLastAppliedConfiguration bool
```